### PR TITLE
Add ignore option to component generation.

### LINKED ISSUE #486: Allow excluding file patterns when generating component definitions
Thanks a lot for making Dash -- we have found it very useful for building dashboards.

We ran into an issue migrating to the new `dash-generate-components` utility.

In [dash-bootstrap-components](https://github.com/ASIDataScience/dash-bootstrap-components), we keep the components in `src/components`. Besides components files, this also includes a few unit test files called, e.g. `src/components/__tests__/DropdownMenu.test.js`. When we run `dash-generate-components ./src/components <output>`, these test files also get picked up. This leads to a traceback in the build logs: 

```
Error with path src/components/__tests__/DropdownMenu.test.jsError: No suitable component definition found.
Error: No suitable component definition found.
at parse (/project/pascal/dash-bootstrap-components/node_modules/react-docgen/dist/parse.js:84:9)
at Object.defaultParse [as parse] (/project/pascal/dash-bootstrap-components/node_modules/react-docgen/dist/main.js:66:30)
at parseFile (/opt/anaconda/envs/Python3/lib/python3.6/site-packages/dash/extract-meta.js:64:51)
at dirs.forEach.filename (/opt/anaconda/envs/Python3/lib/python3.6/site-packages/dash/extract-meta.js:84:17)
at Array.forEach (<anonymous>)
at collectMetadataRecursively (/opt/anaconda/envs/Python3/lib/python3.6/site-packages/dash/extract-meta.js:79:14)
at dirs.forEach.filename (/opt/anaconda/envs/Python3/lib/python3.6/site-packages/dash/extract-meta.js:82:17)
at Array.forEach (<anonymous>)
at collectMetadataRecursively (/opt/anaconda/envs/Python3/lib/python3.6/site-packages/dash/extract-meta.js:79:14)
at componentPaths.forEach.componentPath (/opt/anaconda/envs/Python3/lib/python3.6/site-packages/dash/extract-meta.js:15:5)
```

While the error is, of course, legitimate, it would be nice to be able to suppress these tracebacks. I can see three routes:
- do nothing -- after all, this doesn't stop us from building `metadata.json`, it just makes it slightly more confusing.
- in `dash/extract-meta.js`, we could explicitly blacklist `__tests__` directories in the same way that files that don't end in `jsx?` are black-listed. AFAICT, the `__tests__` directory structure is the default with [jest](https://jestjs.io/docs/en/configuration.html#testmatch-array-string). [react-docgen](https://github.com/reactjs/react-docgen#cli) ignores `node_modules`, `__tests__` and `__mocks__` by default, so there is definitely a precedent.
- add a `--ignore` argument to `dash-generate-components` that allows passing file globs to be excluded. 

Very happy to submit a PR if you decide on what the best course of action is.

### PR DESCRIPTION
**Pull request by T4rk1n**
Adds an ignore option to `dash-generate-components` cli, default is `^_` so it will ignore the `__tests__` folder pattern by default.

Closes #486

### CODE REVIEW

File: dash/extract-meta.js [hunk 1]
```diff
@@ -36,8 +42,7 @@ function writeError(msg, filePath) {
 }
 
 function checkWarn(name, value) {
-    const excluded = ['setProps', 'id', 'className', 'style', 'dashEvents', 'fireEvent'];
-    if (value.length < 1 && !excluded.includes(name)) {
+    if (value.length < 1 && !excludedDocProps.includes(name.split('.').pop())) {
```
[the remark this answers is not in the document]
Comment (T4rk1n) [reply dash/extract-meta.js]: The name is actually the Component name + prop name and we only want the prop name.

File: dash/development/component_generator.py [hunk 2]
```diff
@@ -22,14 +22,17 @@ class _CombinedFormatter(argparse.ArgumentDefaultsHelpFormatter,
 
 # pylint: disable=too-many-locals
 def generate_components(components_source, project_shortname,
-                        package_info_filename='package.json'):
+                        package_info_filename='package.json',
+                        ignore='^_'):
```
Comment (Marc-Andre-Rivet) [inline dash/development/component_generator.py]: This covers the `__tests__` case mentioned in https://github.com/plotly/dash/issues/486 but maybe a small set of defaults (e.g `['__tests__, 'tests']) to cover both the dash-bootstrap-components and our own dash-component-boilerplate case would make sense.

Comment (Marc-Andre-Rivet) [reply dash/development/component_generator.py]: This is an additional feature and should be documented in the changelog and the version be bumped.

File: dash/extract-meta.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Marc-Andre-Rivet) [reply dash/extract-meta.js]: Kk thx

**Pull request merged**